### PR TITLE
Feature/switch hiking direction

### PR DESCRIPTION
**Pull request by jakubprogramming**
We can now use a button to switch the hiking direction from within the route_preview page.
Furthermore a starting/ending Point is drawn onto the map.

Showing the hiking direction is done using green (start) and red (end) blurred lines on top of the path. This may be improved later on. Any ideas are very welcome. At the moment I think it will do the job just fine.

### CODE REVIEW

File: lib/components/map_widget.dart [hunk 1]
```diff
@@ -139,6 +142,26 @@ class MapWidgetState extends State<MapWidget> {
     if (!widget.isStatic) initUpdateRouteTimer();
   }
 
+  drawRouteStartingPoint(HikingRoute route){
```
Comment (dephiloper) [inline lib/components/map_widget.dart]: add an return type

File: lib/components/map_widget.dart [hunk 2]
```diff
@@ -139,6 +142,26 @@ class MapWidgetState extends State<MapWidget> {
...
+    mapController.clearCircles();
+    LatLng startingPoint = route.path[0];
+    CircleOptions optionsStartingPoint = CircleOptions(geometry: startingPoint, circleColor: "Red", circleRadius: 12, circleStrokeWidth: 7, circleStrokeColor: "Blue", circleBlur: 0.25, circleOpacity: 1);
+    mapController.addCircle(optionsStartingPoint);
+  }
+
+  drawHikingDirection(HikingRoute route) async{
```
Comment (dephiloper) [inline lib/components/map_widget.dart]: add an return type

**Review by dephiloper — APPROVED** [review]
Nothing to complain about, besides the missing return types.
Visually I think it is necessary to discuss this in more depth because, imo it does not explain itself. I research on some hiking apps for a better visualization.

Approved, and when return types are changed, this can also be merged.

**Comment by jakubprogramming** [comment]
Changes have been applied. Doing the merge now

**Pull request merged**